common_types/src/row/contiguous.rs

```diff
@@ -50,47 +65,197 @@ pub trait ContiguousRow {
     fn datum_view_at(&self, index: usize) -> DatumView;
 }
 
-pub struct ContiguousRowReader<'a, T> {
+struct Encoding;
+
+/// Here is the encoding of the continuous row. The encoding format depends on
+/// whether the row contains null columns:
+///
+/// # Row with null columns
+/// This row will be encoded as:
+/// ```plaintext
+/// +------------------+-----------------+-------------------------+-------------------------+
+/// | num_bits(u32)    |  nulls_bit_set  | datum encoding block... | var-len payload block   |
+/// +------------------+-----------------+-------------------------+-------------------------+
+/// ```
+/// The first block is the number of bits of the `nulls_bit_set`, which is used
+/// to rebuild the bit set. The `nulls_bit_set` is used to record which columns
+/// are null. With the bitset, any null column won't be encoded in the following
+/// datum encoding block.
+///
+/// As for the datum encoding block, most type shares the similar pattern:
+/// ```plaintext
+/// +--------+----------------+
+/// |type(1B)| payload/offset |
+/// +--------+----------------+
+/// ```
+/// If the type has a fixed size, the data payload will follow the data type.
+/// Otherwise, a offset in the var-len payload block pointing the real payload
+/// follows the type.
+///
+/// Here is the payload for variable length type:
+/// ```plaintext
+/// +-----------------+-------------------+
+/// |  var_len(u32)   |   var payload...  |
+/// +-----------------+-------------------+
+/// ```
+///
+/// # Row without null columns
```

Comment:
It seems this layout is same with above, so in fact we have only one layout, this is a special case when bit number is 0.

If my understanding is right, I think we can delete `Row without null columns` section, it feels there are two different layout in current writing style, maybe a note is enough, something like this:
> When there is no null value in row, then num_bits is 0, and bit_set is empty.